src/main/kotlin/com/github/avrokotlin/avro4k/serializer/BigDecimalSerializer.kt

```diff
@@ -16,68 +15,58 @@ import org.apache.avro.LogicalTypes
 import org.apache.avro.Schema
 import org.apache.avro.SchemaBuilder
 import org.apache.avro.generic.GenericFixed
-import org.apache.avro.util.Utf8
 import java.math.BigDecimal
-import java.math.RoundingMode
 import java.nio.ByteBuffer
 import kotlin.reflect.jvm.jvmName
 
+/**
+ * A BigDecimal is serialized in 3 ways:
+ * - as a [String] when schema type is [Schema.Type.STRING] (scale and precision have not effect here)
+ * - as a [ByteBuffer] when schema type is [Schema.Type.BYTES] (see [Conversions.DecimalConversion.toBytes])
+ * - as a [GenericFixed] when schema type is [Schema.Type.FIXED] (see [Conversions.DecimalConversion.toFixed])
+ */
 @OptIn(ExperimentalSerializationApi::class)
-@Serializer(forClass = BigDecimal::class)
```

Comment:
I always thought this would be mandatory, but it seems that I've not correctly understood the documentation. Thanks for removing it!